### DISCUSSION TITLE
Importer from TFCloud failling

### DISCUSSION THREAD

**Discussion by klinux** [terrakube-io/terrakube#958] [Q&A] [answered]
I'm trying to import workspaces from Terraform Cloud and the import failling with error:

`Failed to import workspace: AxiosError: Request failed with status code 404`

The import was working in Terrakube 2.20.0 after update 2.21.0 the import failling.

On the browser, after click to import,  I get this error:

```
{
    "timestamp": "2024-06-11T20:21:14.822+00:00",
    "status": 404,
    "error": "Not Found",
    "path": "/importer/tfcloud/workspaces/"
}
```

This a bug or I make some mistakes on my version upgrade?

Thank you

**Comment by alfespa17** [comment]
I think it should be a bug...... let me do some small test to validate

**Reply by klinux** [reply]
It's looks that / in the end of path, is the cause of 404 status code

**Reply by alfespa17** [reply]
Can you share some screenshot? I am testing the page and for me it is loading.

![image](https://github.com/AzBuilder/terrakube/assets/4461895/f003814c-4448-409e-9b3f-e9fd5c2fc975)

**Reply by klinux** [reply]
Listing workspaces works, the problem is when I trying to import.

**Reply by alfespa17** [reply]
You are right
![image](https://github.com/AzBuilder/terrakube/assets/4461895/68de7eb4-fb15-4f66-a506-3a4ff820790c)

**Reply by klinux** [reply]
It's looks that / in the end of path, is the cause of 404 status code, in the UI the code is:

```
const importWorkspace = async (workspace) => {
    try {
      const response = await axiosInstance.post(
        `${
          new URL(window._env_.REACT_APP_TERRAKUBE_API_URL).origin
        }/importer/tfcloud/workspaces/`,
```

It's look that slash in the end, is the problem.

**Reply by klinux** [reply]
Yaaa, the slash is the problem, I made request directly without slash in the end, and api works to import.

**Reply by alfespa17** [reply]
Do you want to send the pull request with the fix?

**Reply by klinux** [reply]
https://github.com/AzBuilder/terrakube/pull/960

**Reply by alfespa17** [reply, marked as answer]
Fixed in the following version

https://github.com/AzBuilder/terrakube/releases/tag/2.21.2
